cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/SourceControlManagementService.java

```diff
@@ -57,8 +59,14 @@ private NamespaceTable getNamespaceTable(StructuredTableContext context) throws
     return new NamespaceTable(context);
   }
 
-  public void setRepository(NamespaceId namespace, RepositoryConfig repository) throws NamespaceNotFoundException {
+  public void setRepository(NamespaceId namespace, RepositoryConfig repository)
+    throws NamespaceNotFoundException, InvalidRepositoryConfigException {
     accessEnforcer.enforce(namespace, authenticationContext.getPrincipal(), StandardPermission.UPDATE);
+    if (repository == null) {
+      throw new InvalidRepositoryConfigException("Repository configuration cannot be null.");
+    }
+    
+    repository.validate();
```

Comment:
validate should be called in the handler and not in here, as it is just checking basic things like required fields are not null. It's similar to how we check the namespace character set in the handler.

Reply:
moved to the handler